src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +40,342 @@ const ActorInit En_Door_InitVars = {
+                }
+            }
+        } else if (func_800A56C8(&this->skelAnime, sDoorAnimCloseFrames[this->unk_190])) {
+            // Play the appropriate sound on close
+            Audio_PlayActorSound2(&this->actor,
+                                  (globalCtx->sceneNum == SCENE_HAKADAN || globalCtx->sceneNum == SCENE_HAKADANCH ||
+                                   globalCtx->sceneNum == SCENE_HIDAN)
+                                      ? NA_SE_EV_IRON_DOOR_CLOSE
+                                      : NA_SE_EV_DOOR_CLOSE);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC8F4.s")
+void EnDoor_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDoor* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC928.s")
+s32 EnDoor_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                            Actor* thisx) {
+    s32 pad;
+    TransitionActorEntry* transitionEntry;
+    Gfx** temp_a2;
+    s32 pad2;
+    s16 phi_v0_2;
+    s32 phi_v0;
+    EnDoor* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC998.s")
+    if (limbIndex == 4) {
+        temp_a2 = &D_809FCEE4[(s8)this->displaylistIdx];
+        transitionEntry = &globalCtx->transitionActorList[(u16)this->actor.params >> 0xA];
+        rot->z += this->actor.posRot.rot.y;
+        if ((globalCtx->roomCtx.prevRoom.num >= 0) ||
+            (transitionEntry->sides[0].room == transitionEntry->sides[1].room)) {
+            phi_v0_2 = ((this->actor.shape.rot.y + this->skelAnime.limbDrawTbl[3].z) + rot->z) -
+                       Math_Vec3f_Yaw(&globalCtx->view.eye, &this->actor.posRot.pos);
+            *dList = (ABS(phi_v0_2) < 0x4000) ? temp_a2[0] : temp_a2[1];
+        } else {
+            phi_v0 = this->unk_192;
+            if (transitionEntry->sides[0].room != this->actor.room) {
+                phi_v0 ^= 1;
+            }
+            *dList = temp_a2[phi_v0];
+        }
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FC9DC.s")
+void EnDoor_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDoor* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Update.s")
+    if (this->actor.objBankIndex == this->requiredObject) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_door.c", 910);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/func_809FCC3C.s")
+        func_80093D18(globalCtx->state.gfxCtx);
+        SkelAnime_Draw(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, EnDoor_OverrideLimbDraw, NULL,
+                       &this->actor);
+        if (this->actor.posRot.rot.y != 0) {
+            if (!this) {}
```

Comment:
```suggestion
            if(1) {}
```
It doesn't have to be `this`.